sa/model_test.go

```diff
@@ -233,6 +185,23 @@ func TestAuthzModel(t *testing.T) {
 	if authzPBOut.Challenges[0].Validationrecords[0].Port != "443" {
 		test.Assert(t, false, fmt.Sprintf("rehydrated http-01 validation record expected port 443 but found %v", authzPBOut.Challenges[0].Validationrecords[0].Port))
 	}
+
+	authzPB = newTestAuthzPB(clk.Now())
+	authzPB.Identifier = identifier.NewIP(netip.MustParseAddr("1.2.3.4")).ToProto()
+	authzPB.Challenges[0].Validationrecords[0].Url = "https://1.2.3.4"
+	authzPB.Challenges[0].Validationrecords[0].Hostname = "1.2.3.4"
```

Comment:
Here you're specifically setting some fields in the ValidationRecords but not checking them. I think it would make sense to check them.